src/main/java/address/controller/RootLayoutController.java

```diff
@@ -56,13 +75,50 @@ public void setShortcuts(){
         menuFileSaveAs.setAccelerator(ShortcutsManager.SHORTCUT_FILE_SAVE_AS);
     }
 
+    @Subscribe
+    private void handleSaveLocationChangedEvent(SaveLocationChangedEvent e) {
+        updateSaveLocDisplay();
+    }
+
+    @Subscribe
+    private void handleMirrorLocationChangedEvent(MirrorLocationChangedEvent e) {
+        updateMirrorLocDisplay();
+    }
+
+    private void updateSaveLocDisplay() {
+        saveLocText.setText(SAVE_LOC_TEXT_PREFIX + (PrefsManager.getInstance().isSaveLocationSet() ?
+                PrefsManager.getInstance().getSaveLocation().getName() : LOC_TEXT_NOT_SET));
+    }
+
+    private void updateMirrorLocDisplay() {
+        mirrorLocText.setText(MIRROR_LOC_TEXT_PREFIX + (PrefsManager.getInstance().isMirrorLocationSet() ?
+                PrefsManager.getInstance().getMirrorLocation().getName() : LOC_TEXT_NOT_SET));
+    }
+
+    /**
+     * @return a file chooser for choosing xml files. The initial folder is set to the same folder that the
+     *     current data file is located (if any).
+     */
+    private FileChooser getXmlFileChooser() {
+        // Set extension filter
+        final FileChooser.ExtensionFilter extFilter = new FileChooser.ExtensionFilter("XML files (*.xml)", "*.xml");
+        final FileChooser fileChooser = new FileChooser();
+
+        fileChooser.getExtensionFilters().add(extFilter);
+        final File currentFile = PrefsManager.getInstance().getSaveLocation();
+        fileChooser.setInitialDirectory(currentFile.getParentFile());
+        return fileChooser;
+    }
+
+
     /**
      * Creates a new empty address book at the default filepath
      */
     @FXML
     private void handleNew() {
-        PreferencesManager.getInstance().setPersonFilePath(null);
-        modelManager.resetData(Collections.emptyList(), Collections.emptyList());
+        PrefsManager.getInstance().clearSaveLocation();
+        PrefsManager.getInstance().clearMirrorLocation();
+        modelManager.clearModel();
```

Comment:
It seems very readable to me, and there is nowhere else that reuses this code right now.

Can't come up with a good name for the extracted method :(
Need to find a reasonable abstraction that covers these 3 steps for the name.